Main/inc/Analysis.hh

```diff
@@ -101,52 +107,159 @@ namespace roofitter {
+        for (unsigned int c = 0; c < branchleaf.length(); ++c)
+        {
+            if (branchleaf[c] == '.')
+            {
+                branchleaf[c] = '/';
+                branch_name = branchleaf.substr(0, c);
+                leaf_name = branchleaf.substr(c+1, branchleaf.length());
+            }
+        }
+
+        tree_flat->Branch(leaf_name.c_str(), &var, (leaf_name+"/F").c_str());
+        leaf = tree->GetLeaf(branchleaf.c_str());
+
+        // Formula for the cut to apply on the tree
+        TTreeFormula *cut_formula = new TTreeFormula("cut", cutcmd(), tree);
+
+        Int_t n_entries = (Int_t) tree->GetEntries();
+        for (Int_t i = 0; i < n_entries; i++)
+        {
+            tree->GetEntry(i);
+            if (cut_formula->EvalInstance() == 1)
+            {
+                var = leaf->GetValue();
+                tree_flat->Fill();
+            }
+        }
+
+/*
+        std::vector<std::string> branchleaf;
+        std::vector<Float_t> vars(_observables.size());
+        std::vector<TLeaf *> leaves;
+        TFile *file = new TFile("trkana_flat.root", "RECREATE");
+        TTree *tree_flat = new TTree("trkana_flat", "flatten trkana tree");
+
+        for (unsigned int i_obs = 0; i_obs < _observables.size(); ++i_obs)
+        {
+            auto& obs_conf = _observables[i_obs].getConf();
+            branchleaf.push_back(obs_conf.leaf());
+            std::string branch_name, leaf_name;
+            // Replace the "." by a "/" in the string to create the leaf later; also extract the name of the branch and the name of the leaf
+            for (unsigned int c = 0; c < branchleaf[i_obs].length(); ++c)
+            {
+                if (branchleaf[i_obs][c] == '.')
+                {
+                    branchleaf[i_obs][c] = '/';
+                    branch_name = branchleaf[i_obs].substr(0, c);
+                    leaf_name = branchleaf[i_obs].substr(c+1, branchleaf[i_obs].length());
+                }
+            }
+
+            tree_flat->Branch((branch_name+leaf_name).c_str(), &vars[i_obs], (branch_name+leaf_name+"/F").c_str());
+            leaves.push_back(tree->GetLeaf(branchleaf[i_obs].c_str()));
+        }
+
+        // Formula for the cut to apply on the tree
+        TTreeFormula *cut_formula = new TTreeFormula("cut", cutcmd(), tree);
+
+        std::cout << "size vars: " << vars.size() << std::endl;
+        Int_t n_entries = (Int_t) tree->GetEntries();
+        for (Int_t i = 0; i < n_entries; i++)
+        {
+            for (unsigned int i_obs = 0; i_obs < _observables.size(); ++i_obs)
+            {
+                tree->GetEntry(i);
+                if (cut_formula->EvalInstance() == 1)
+                {
+                    std::cout << "index: " << i_obs << std::endl;
+                    vars[i_obs] = leaves[i_obs]->GetValue();
+                    tree_flat->Fill();
+                }
+            }
+        }
+*/
```

Comment:
Do we need to keep this commented out block? Does it have code that we might want to use in future?

Reply:
Yes, this is the code I wrote to have multiple observables, but there is a memory leak that I haven't fixed yet. Now that the unbinned fit is implemented, I am going to work on the 2D fit so I'm going to solve this issue very soon.

Reply:
That's fair enough :)